lib/internal/http2/core.js

```diff
@@ -344,6 +348,31 @@ function emit(self, ...args) {
   ReflectApply(self.emit, self, args);
 }
 
+function createServerStream(session, handle, id, options, headers, endOfStream) {
+  // eslint-disable-next-line no-use-before-define
+  const stream = new ServerHttp2Stream(session, handle, id, options, headers);
+  if (endOfStream) {
+    stream.push(null);
+    stream[kState].endAfterHeaders = true;
+  }
+
+  if (headers[HTTP2_HEADER_METHOD] === HTTP2_METHOD_HEAD) {
+    // For head requests, there must not be a body...
+    // end the writable side immediately.
+    stream.end();
```

Comment:
shouldn't it be inside an `else if` instead? What if we receive an `endOfStream` = true and HTTP2_HEADER_METHOD = head?

Reply:
Maybe, to be honest I've just kept the same logic it was implemented before.